docs/thirdparty-rest-v1.0-OpenApi.yaml

```diff
@@ -0,0 +1,846 @@
+              "requestId": "456",
+              "initiatorId": "PISP A",
+              "participantId": "DFSP A",
+              "scopes": [
+                { 
+                  "scope": "account.balanceInquiry",
+                  "accountIdentifier": "12345-67890"
+                },
+                { 
+                  "scope": "account.sendTransfer",
+                  "accountIdentifier": "12345-67890"
+                },
+                { 
+                  "scope": "account.sendTransfer",
+                  "accountIdentifier": "77777-38937"
+                }
+              ],
+              "credential": null
+            }
+      responses:
+        202:
+          $ref: '#/responses/Response202'
+        400:
+          $ref: '#/responses/ErrorResponse400'
+        401:
+          $ref: '#/responses/ErrorResponse401'
+        403:
+          $ref: '#/responses/ErrorResponse403'
+        404:
+          $ref: '#/responses/ErrorResponse404'
+        405:
+          $ref: '#/responses/ErrorResponse405'
+        406:
+          $ref: '#/responses/ErrorResponse406'
+        501:
+          $ref: '#/responses/ErrorResponse501'
+        503:
+          $ref: '#/responses/ErrorResponse503'
+  
+  /consent/{ID}:
+    put:
+      # TODO: better description
+      description: >
+        Updates a consent object.
+      summary: UpdateConsent
+      operationId: UpdateConsent
+      tags:
+        - consent
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/ID'
+        - name: body
+          in: body
+          required: true
+          schema: 
+            $ref: '#/definitions/Consent'
+      # Headers
+        - $ref: '#/parameters/Content-Length'
+        - $ref: '#/parameters/Content-Type'
+        - $ref: '#/parameters/Date'
+        - $ref: '#/parameters/X-Forwarded-For'
+        - $ref: '#/parameters/FSPIOP-Source'
+        - $ref: '#/parameters/FSPIOP-Destination'
+        # TODO: do we need encryption on this?
+        - $ref: '#/parameters/FSPIOP-Encryption'
+        - $ref: '#/parameters/FSPIOP-Signature'
+        - $ref: '#/parameters/FSPIOP-URI'
+        - $ref: '#/parameters/FSPIOP-HTTP-Method'
+      x-examples:
+          "application/json":
+            { 
+              "requestId": "456",
+              "initiatorId": "PISP A",
+              "participantId": "DFSP A",
+              "scopes": [
+                {
+                  "scope": "account.balanceInquiry",
+                  "accountIdentifier": "12345-67890"
+                },
+                { 
+                  "scope": "account.sendTransfer",
+                  "accountIdentifier": "12345-67890"
+                },
+                { 
+                  "scope": "account.sendTransfer",
+                  "accountIdentifier": "77777-38937"
+                }
+              ],
+              "credential": {
+                "id": "5678", 
+                "credentialType": "FIDO",
```

Comment:
is there a need for `credential` object to having properties with `credential` prefix ? -> maybe `credential.type` instead `credential.credentialType`

Reply:
same for `credentialStatus`...